linkml/generators/fhirgen.py

```diff
@@ -0,0 +1,393 @@
+        # TODO: This is super temporary; has to do with unit test and not all enums being well formed:
+        if 'EmploymentEventType' in enums:
+            enums = {'EmploymentEventType': enums['EmploymentEventType']}
+        # todo: `en` and `n` below are redundant?
+        for en, e in enums.items():
+            n = self.name(e)
+            d: Dict = self.serialize_enum(n, e)
+            out_str: str = json.dumps(d)
+            if outpath:
+                self._write(out_str, outpath, n)
+
+    @staticmethod
+    def model_mappings(path: str = MODEL_MAPPING_PATH) -> Dict[str, str]:
+        """Get model mappings from CSV"""
+        with open(path) as csvfile:
+            reader = csv.reader(csvfile, delimiter=',')
+            rows = [x for x in reader][1:]  # don't need header
+        d = {
+            row[0]: row[1]  # easiest way; assumes linkml is always col 1, and fhir is col 2
+            for row in rows
+        }
+        d = {k: v for k, v in d.items() if k}  # removes anything left over from empty CSV rows
+
+        return d
+
+    # TODO: Most of the work should go here
+    # todo: Since I want to return a dict, should I call this dictify instead of serialize?
+    def serialize_enum(self, enum_name: str, enum: EnumDefinition) -> Dict:
+        """
+        Serialize Enums in schema to JSON
+
+        todo: use enum_name if needed
+        :param enum_name: The enum name
+        :param enum: The enum
+        :return:
+        """
+        d = {}
+        mappings: Dict[str, str] = self.model_mappings()
+
+        # TODO: flesh this out
+        for linkml_field, fhir_field in mappings.items():
+            # TODO: What if the enum doesn't have the field defined?
+            linkml_val = getattr(enum, linkml_field)
+            if linkml_field in FHIR_VALUESET_FIELD_MUTATIONS:
+                fhir_val = FHIR_VALUESET_FIELD_MUTATIONS[linkml_field](linkml_val)
+            else:
+                fhir_val = linkml_val
+            d[fhir_field] = fhir_val
+
+        return d
+
+    def _write(self, out_str: str, outpath: str, name: str) -> None:
+        """
+        Writes JSON to outpath
+
+        :param out_str: string to be written
+        :param outpath: location
+        :param name: base name - should correspond to element name
+        :return:
+        """
+        path = Path(outpath)
+        path.mkdir(parents=True, exist_ok=True)
+        file_name = f"{name}.{self._file_suffix()}"
+        # TODO: Change this to use `json` package
+        with open(path / file_name, "w", encoding="UTF-8") as stream:
+            stream.write(out_str)
+
+    def schema_title(self) -> str:
+        """
+        Returns the title of the schema.
+
+        Uses title field if present, otherwise name
+
+        :return:
+        """
+        s = self.schemaview.schema
+        if s.title:
+            return s.title
+        else:
+            return s.name
+
+    def _file_suffix(self):
```

Comment:
Not sure if I should keep this or not, since I'm only doing `json` now, and both `json` and `xml` have the same spellings for their file extensions.

Reply:
is there an equivalent of the python linkml stack for fhir? e.g. where you can instantiate objects and let the framework take care of serialization

Reply:
Not that I'm aware of yet; not like a general purpose library.

This Smart-on-FHIR (mainly something for web apps) client is the closest I've seen that has such models. Here's its [ValueSet](https://github.com/smart-on-fhir/client-py/blob/master/fhirclient/models/valueset.py). I think I might have seen Harold work on something familiar. Not sure how well this is maintained. Anyway looking at that particular example, it looks like it accepts a `jsondict`, not named args. And I don't see how it's setting values from that dict; looks like somewhere in the superclass calls.

Maybe something I can ask on the FHIR Zulip channel, if there's any more general purpose FHIR model library.